dockstore-integration-testing/src/test/java/io/dockstore/client/cli/GA4GHV2FinalIT.java

```diff
@@ -401,11 +401,49 @@ void toolsIdGet4Workflows() throws Exception {
 
     @Test
     void testGetHeaderLinksContainFilters() throws Exception {
-        Response response = checkedResponse(baseURL + "tools?toolClass=Tool&limit=1");
+        // Insert the 4 workflows into the database using migrations
+        CommonTestUtilities.setupSamePathsTest(SUPPORT);
+
+        Response response = checkedResponse(baseURL + "tools?toolClass=Workflow&limit=1");
         MultivaluedMap<String, Object> headers = response.getHeaders();
-        assertTrue(headers.get("self_link").toString().contains("toolClass=Tool"));
-        assertTrue(headers.get("last_page").toString().contains("toolClass=Tool"));
-        assertTrue(headers.get("next_page").toString().contains("toolClass=Tool"));
+        assertTrue(headers.get("self_link").toString().contains("toolClass=Workflow"));
+        assertTrue(headers.get("last_page").toString().contains("toolClass=Workflow"));
+        assertTrue(headers.get("next_page").toString().contains("toolClass=Workflow"));
+
+        // reset DB for other tests
+        CommonTestUtilities.dropAndCreateWithTestData(SUPPORT, false);
```

Comment:
This pattern was here, but is the idea that most of the tests are read-only and don't need to initialize the DB if it's not changing? And only reset the DB if it does change?

Reply:
Yes on both.